src/main/java/crawlercommons/fetcher/http/BaseHttpFetcher.java

```diff
@@ -109,4 +112,8 @@ public RedirectMode getRedirectMode() {
         return _redirectMode;
     }
 
+    public void setProxy(String scheme, String host, int port) {
+        this._proxy = new HttpHost(host, port, scheme);
```

Comment:
No need for `this.`, which is one of the reasons we use a leading `_`.

Also would be good to have a `getProxy()` call.